### ISSUE TITLE
Optional minimum number of student classes

### ISSUE THREAD

**Issue by milescalabresi** [learning-unlimited/ESP-Website#2535]
I'm not sure if this is something I fully support, but I wanted at least to open it to discussion after multiple Chapters have independently wandered into asking about this possibility.

For a little bit of backstory, many programs have wondered how they can increase the yield of students who actually attend of those who register. I've looked at some data across Splashes, and the strongest known factor that correlates with attendance is number of classes. Students who sign up for a full day are well above the average, meanwhile students who sign up for one class attend at an average rate in the teens.
Towards that end, some admins have discussed instituting minimum requirements. There are a number of  can make a class selection process where you don’t enroll before hitting a submit button (which would allow for admins to *optionally* institute a minimum requirement on number of classes), but this change would very possibly cause confusion and create large numbers of students who thought they had registered, so it would have to be very obvious that they were in "browsing" mode, not "submitting" mode.

I was thinking this would be an optional feature so that anyone who wanted to continue with current interface could do so, but there's also a case to be made for switching all sites to a select-then-submit format for consistency so that sites who use the minimum requirement feature aren't penalized.

On the other hand, it seems very imposing to make students sign up for classes. Obviously, we do not want to force students to sign up for classes that don't interest them. However, the argument can be made that if the minimum is only 2-3 classes, then since 1-class students already don't attend (and, in fact, a majority of those who do later add a class onsite), there is unlikely to be anyone who is forced into going to a class they dislike.

Thoughts?

**Comment by willgearty** [comment]
One could make the argument that this functionality is already the expectation for many students (select, then submit). We've gotten innumerable emails asking how students should go about "submitting" their class selections (or even lottery preferences). #1166 and #1437 have discussed this before, although they are specifically about the two-phase lottery. I'm not necessarily saying I'm in support of this, rather that it wouldn't be the craziest thing.
Furthermore, is the problem that students are taking up spots from other students at these programs? Or that the programs just see larger numbers of students registered than actually show up, and so maybe they ordered too much food, etc.?

**Comment by milescalabresi** [comment]
Good point. I think my main fear is that the one student out of ten who doesn't realize they have to would register in good faith then show up only to realize they're not registered.

To answer your question, yes, all of the above, plus the fact that it leads to really screwy counts for smaller classes when a nontrivial percentage of the students may or may not show up.

**Comment by willgearty** [comment]
@milescalabresi in terms of smaller class enrollment, I believe there's a way to allow for over-enrollment, in case that could help?
Also, I wonder, are the students not coming because they don't have many classes, or do they not have many classes because they don't really plan to come in the first place, so they don't register for many classes? If it's the former, I think something like #2425 should help encourage more enrollment (although that pull request is lottery-focused). If it's the latter, then maybe something like what you are suggesting would be good.

**Comment by milescalabresi** [comment]
Yes, but over-enrollment doesn't solve the issue for teachers of the big gaps between expected (and therefore number of supplies needed) and actual number of participants.

That's a good question. I don't know the answer. They're also not mutually exclusive options. The limited evidence we have is that when you bother students with a small number of classes, slightly more of them sign up for more classes than drop their existing classes, but there are some in each category. Unfortunately, I don't know of any data that looks at attendance rates among kids who previously had a small number of classes but signed up for more. In any case, the bottom line seems to be that kids have a lot of inertia and reminding them to do something is better than nothing. #thoughts

**Comment by Dhirenderchoudhary** [comment]
@willgearty please assign me this issue as this was part of my previous issue assigned so can fix both.
